sky/provision/kubernetes/instance.py

```diff
@@ -876,6 +876,17 @@ def _create_resource_thread(i: int):
             pod_spec_copy['spec']['tolerations'] = existing_tolerations + [
                 tpu_toleration
             ]
+        if needs_gpus:
+            gpu_toleration = {
+                'key': kubernetes_utils.get_gpu_resource_key(),
+                'operator': 'Exists',
+                'effect': 'NoSchedule'
+            }
+            # Preserve existing tolerations if any
+            existing_tolerations = pod_spec_copy['spec'].get('tolerations', [])
+            pod_spec_copy['spec']['tolerations'] = existing_tolerations + [
+                gpu_toleration
+            ]
```

Comment:
Add comment for why we need this, but not before.

Reply:
Updated

Reply:
Is it ok for non-DWS case?

Reply:
Yes, the toleration just ensure that the Pod can be scheduled to the node with this toleration, and if the node does not have this toleration, the Pod still can be scheduled to it.